### ISSUE TITLE
warning: failed prop type: invalid prop 'ImageComponent' of type 'object' supplied to 'Avatar', expected 'function

### ISSUE THREAD

**Issue by tuhintechvill** [react-native-training/react-native-elements#1472]
When I add leftAvatar a warning shows in android emulator
> warning: failed prop type: invalid prop 'ImageComponent' of type
> 'object' supplied to 'Avatar', expected 'function

react-native-elements version : "^1.0.0-beta6" My code:
```
<FlatList data={this.state.data} renderItem={(data) => (
       <View>
           <ListItem
                        roundAvatar
                        rightIcon={{ name: 'chevron-right',type: 'MaterialIcon'}}
                        title={'Tuhin Hossain'}
                        subtitle={'Software Engineer'}
                        leftAvatar={{ 
                            title: 'Tuhin Hossain',
                            source: { uri: "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg"} }}
           />
                    <Divider style={{ backgroundColor: 'slategray' }} />
         </View>

)} keyExtractor={(value, index) => value.id}/>
```

**Comment by iRoachie** [comment]
What version of react-native-elements are you using? There's a reason we have an issue template

**Comment by tuhintechvill** [comment]
@iRoachie my react-native-elements version is :  "^1.0.0-beta6"

**Comment by haruelrovix** [comment]
@tuhintechvill I can reproduce the issue using **^1.0.0-beta7**.

<img src="https://user-images.githubusercontent.com/17120764/47056639-8a22ff00-d1e7-11e8-914a-c6b6d85c06e4.png" width=250 />

@iRoachie here is the repo in case you're interested. [rne-issue-1472](https://bitbucket.org/haruelrovix/rne-issue-1472/src/master/)

**Comment by Monte9** [comment]
damn.. that yellow warning box looks so much nicer! 😍
What RN version is that?

**Comment by iRoachie** [comment]
I see the warning, however i’m not sure why this comes up. The Image component from react native is a function no?

**Comment by haruelrovix** [comment]
@iRoachie Nope. [`context`](https://github.com/react-native-training/react-native-elements/blob/15b99529b5b389ab9dfcada1134a47dfd904525d/src/config/withTheme.js#L25) inside `ThemeConsumer` returns a `WrappedComponent` which is an object.
```js
  const notClassComponent = <WrappedComponent {...props} />;
  console.log(typeof notClassComponent);

  return isClassComponent(WrappedComponent) ? (
    <WrappedComponent ref={forwardedRef} {...props} />
  ) : (
    notClassComponent
  );
```

**console**:

![image](https://user-images.githubusercontent.com/17120764/47376351-985ba880-d71c-11e8-8c80-6a6d2bc9e299.png)

**Comment by iRoachie** [comment]
Ahh thank you for debugging that. Can you make a PR for this fix?

**Comment by iRoachie** [comment]
Already fixed @lukmanfarizi, will release in next version. Just look for the `Fixed - Next Release` label.

**Comment by lukmanfarizi** [comment]
oohh, sorry @iRoachie. i didn't notice that.

**Comment by sbefort** [comment]
This is still an issue in 1.07.

**Comment by iRoachie** [comment]
@sbefort It's been fixed and will release in next version which is 1.0.0

**Comment by iRoachie** [comment]
You can use 1.0.0-beta5

**Comment by venkatmithunk** [comment]
@iRoachie  well I am currently using 1.0.0-beta5 :|

**Comment by renato-bvdash** [comment]
I am using 1.0.0-beta7 and still is an issue.

**Comment by iRoachie** [comment]
@renato-bvdash it's fixed on `next` branch and will be available next release

**Comment by darmie** [comment]
This is still a problem, when will the "next" release happen?

**Comment by iRoachie** [comment]
Mid December @darmie

**Comment by lednhatkhanh** [comment]
Nice, facing the same problem, really happy that it's been fixed.

**Comment by oxyii** [comment]
Did not fixed yet [link](https://github.com/react-native-training/react-native-elements/blob/master/src/avatar/Avatar.js#L214)

**Comment by iRoachie** [comment]
@oxyii It's fixed on `next` branch  https://github.com/react-native-training/react-native-elements/blob/next/src/avatar/Avatar.js#L223

**Comment by Faolain** [comment]
Hey @iRoachie any idea when the next release will be ready?

**Comment by iRoachie** [comment]
Hey @Faolain we were to release in December, but after talking with some of the other maintainers, they opted for January where it's less busy.

There are small remaining items left https://github.com/react-native-training/react-native-elements/milestone/6 for the release. However we may remove some of them if it's too complex or too time consuming for right now

**Comment by ngandhy** [comment]
Hi @iRoachie! Just wanted to check-in and see if there was any progress on this. Thanks!

**Comment by Monte9** [comment]
@ngandhy 1.0 is ready for release. We are working on release notes right now and will be releasing it this weekend. Thanks for being patient! 🙏

**Comment by VitorBrangioni** [comment]
Same problem to version '1.0.0-beta'. When it will be available?

**Comment by oxyii** [comment]
@VitorBrangioni fixed in stable releases, in `1.0.0` too. Please upgrade to non-beta version e.g. latest for now `1.1.0`
